packages/components/src/components/flyout/flyout.tsx

```diff
@@ -0,0 +1,225 @@
+    if (this.open) {
+      setScrollLock(true);
+      this.updateFocusTrap(true);
+    }
+
+    getShadowRootHTMLElement(this.host, 'slot').addEventListener('slotchange', () => {
+      if (this.open) {
+        // 1 tick delay is needed so that web components can be bootstrapped
+        setTimeout(() => {
+          this.updateFocusTrap(true);
+          this.dismissBtn.shadowRoot.querySelector('button').focus(); // set initial focus
+        });
+      }
+    });
+  }
+
+  public componentDidRender(): void {
+    if (this.open) {
+      if (this.hasSecondaryContent && this.hasFooter) {
+        this.onScroll();
+      }
+      // Necessary to select button to make :focus-visible work
+      this.dismissBtn.shadowRoot.querySelector('button').focus();
+    }
+  }
+
+  public disconnectedCallback(): void {
+    setFocusTrap(this.host, false);
+    setScrollLock(false);
+  }
+
+  public render(): JSX.Element {
+    validateProps(this, propTypes);
+
+    this.hasHeader = hasNamedSlot(this.host, 'header');
+    this.hasFooter = hasNamedSlot(this.host, 'footer');
+    this.hasSecondaryContent = hasNamedSlot(this.host, 'secondary-content');
+
+    attachComponentCss(
+      this.host,
+      getComponentCss,
+      this.open,
+      this.position,
+      this.hasHeader,
+      this.hasFooter,
+      this.hasSecondaryContent,
+      this.theme
+    );
+
+    const PrefixedTagNames = getPrefixedTagNames(this.host);
+
+    const dismissBtn = (
+      <PrefixedTagNames.pButtonPure
+        class="dismiss"
+        type="button"
+        hideLabel
+        icon="close"
+        theme={this.theme}
+        onClick={this.dismissFlyout}
+        ref={(el) => (this.dismissBtn = el)}
+      >
+        Dismiss flyout
+      </PrefixedTagNames.pButtonPure>
+    );
+    return (
+      <Host onMouseDown={this.onMouseDown}>
+        <div
+          class="root"
+          role="dialog"
+          {...parseAndGetAriaAttributes({
+            'aria-modal': true,
+            'aria-hidden': !this.open,
+            ...parseAndGetAriaAttributes(this.aria),
+          })}
+          tabIndex={-1}
+          ref={(el) => (this.dialog = el)}
+          {...(this.hasSecondaryContent && { onScroll: this.onScroll })} // if no secondary content is used scroll shadows are done via CSS
+        >
+          <div class="header" ref={(el) => (this.header = el)}>
+            {this.hasHeader && (
+              <div class="header-content">
+                <slot name="header" />
+              </div>
+            )}
+            {dismissBtn}
+          </div>
+          <div class="content">
+            <slot />
+          </div>
+          {this.hasFooter && (
+            <div class="footer" ref={(el) => (this.footer = el)}>
+              <slot name="footer" />
+            </div>
+          )}
+          {this.hasSecondaryContent && (
+            <div class="secondary-content">
+              <slot name="secondary-content" />
```

Comment:
Not so sure about the name. Why is a secondary content after the footer?

Reply:
Any suggestions? :)